crates/astria-sequencer/src/assets/state_ext.rs

```diff
@@ -192,30 +196,37 @@ impl<T: ?Sized + StateRead> StateReadExt for T {}
 #[async_trait]
 pub(crate) trait StateWriteExt: StateWrite {
     #[instrument(skip_all)]
-    fn put_native_asset(&mut self, asset: &asset::TracePrefixed) {
-        self.put_raw(NATIVE_ASSET_KEY.to_string(), asset.to_string().into_bytes());
+    fn put_native_asset(&mut self, asset: asset::TracePrefixed) -> Result<()> {
```

Comment:
Intuitively it makes sense for a `put_*` method to take an object by value. On the other hand I notice that where before you could pass the asset by value, you know explicitly `clone` it, but then in the function body use a reference `&asset`.

Given that many other changes in this PR are performance focused, should we revert this diff?

Reply:
The idea is that this PR was updating the state ext APIs ready for the further storage changes.  In the next PR, I'll be introducing the in-mem cache, so we'll need the actual value for insertion there before serialising it if required.